File: problem094.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        IList<int> order = new List<int>();
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode node = root;
        while (node != null) {
            stack.Push(node);
            node = node.left;
            while (node is null && stack.Count > 0) {
                node = stack.Pop();
                order.Add(node.val);
                node = node.right;
            }
        }
        return order;
    }
}

File: su19/problem094.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root == None:
            return []
        stack = [root]
        order = []
        curr = root
        while len(stack) > 0:
            curr = curr.left
            while curr == None:
                if len(stack) == 0:
                    return order
                curr = stack.pop()
                order.append(curr.val)
                curr = curr.right
            stack.append(curr)
        return order
            
